File: 02_framing_names.ts
function createFrame(names: string[]): string {
    const frameChar = '*'
    const longestNameN: number = Math.max(...names.map(n => n.length))
  
    // construct frame
    const frameParts: string[] = []
    frameParts.push('*'.repeat(longestNameN+4));
    for (const name of names) {
        frameParts.push(`* ${name}${" ".repeat(longestNameN-name.length)} *`)
    }
    frameParts.push('*'.repeat(longestNameN+4))
    
    return frameParts.join('\n')
  }

File: 02_framing_names.py
def createFrame(names):
    longestNameN = max(map(lambda n: len(n), names))

    frameParts = []
    frameParts.append("*"*(longestNameN+4))
    for name in names:
        frameParts.append(f"* {name}{" "*(longestNameN - len(name))} *")
    frameParts.append("*"*(longestNameN+4))

    return "\n".join(frameParts)